set.mm

```diff
@@ -179333,6 +179333,73 @@ other components such as a multiplicative operation for rings (and still
   remain a group per our definition).  Thus, every ring is also a group.  This
   allows theorems from more general structures (groups) to be reused for more
   specialized structures (rings) without having to reprove them.
+  Structures are common in mathematics, but in informal (natural language)
+  proofs the details are assumed in ways that we must make explicit.
+
+  There are many other ways to handle structures.  We chose this
+  "extensible structure" approach
+  because (1) this approach results in simpler notation than other
+  approaches we're aware of and (2) it is easier to do proofs with.
+  We cannot use an approach that uses "hidden" arguments; Metamath does
+  not support hidden arguments, and in any case we want nothing hidden.
+  It would be possible to use a categorical approach (e.g., something
+  vaguely similar to Lean's mathlib).
+  However, instances (the chain of proofs that an
+  X is a Y via a bunch of forgetful functors) can
+  cause serious performance problems for automated tooling,
+  and the resulting proofs would be painful to look at
+  directly (in the case of Lean, they are long past the level where
+  people would find it acceptable to look at them directly).
+  Metamath is working under much stricter conditions than this,
+  and it has still managed to achieve about the same level of flexibility.
+
+  To create a substructure of a given exensible structure, you can simply
+  use the multifunction restriction operator for extensible structures ` |`s `
+  as defined in ~ df-ress .
+  This can be used to turn statements about rings into statements about
+  subrings, modules into submodules, etc.  This definition knows nothing
+  about individual structures and merely truncates the ` Base ` set while
+  leaving operators alone; individual kinds of structures will need to
+  handle this behavior, by ignoring operators' values outside the range
+  (like ` Ring ` ), defining a function using the base set and applying
+  that (like ` TopGrp ` ), or explicitly truncating the slot before use
+  (like ` MetSp ` ).
+  So, for example, the equivalent of some (not defined) NNmnd is simply
+  ` ( CCfld |`s NN ) ` , and similarly for all other subsets of ` CC ` ,
+  which has all  the structure we can show applies to it, and this all
+  comes "for free".
+  Should we come up with some new structure in the future that we
+  wish ` CC ` to inherit, then we change the definition of ` CCfld ` ,
+  reprove all the slot extraction theorems, add a new one,
+  and that's it. None of the other downstream theorems have to change.
+
+  The restriction operation ` |`s ` is not very useful for manifolds.
+  Manifold embeddings or immersions are something else entirely; you
+  clearly need a function to perform that mapping so the implicit subset
+  morphism doesn't cut it. In that case, the typical solution would be to
+  have a set ( M ManifoldImmersion N ) for encoding all these details.
```

Comment:
This paragraph doesn't make much sense out of context. I suggest dropping it.

Reply:
Dropped.